### PR TITLE
feat(appmesh): access log format support for app mesh

### PR DESCRIPTION
**Pull request by EdwardXF**
**Implement access logging feature for AppMesh CDK.**
* Access logging feature allows customer to choose between Text and Json format for the virtual nodes and virtual gateways. Customers can also specify patterns supported by Envoy. (https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage) Example CFN template: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appmesh-virtualnode-loggingformat.html

### All Submissions:

* [Y] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [N] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [Y] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [Y] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-appmesh/lib/shared-interfaces.ts [hunk 1]
```diff
@@ -143,28 +143,138 @@ class FileAccessLog extends AccessLog {
...
+        text: this.text,
+      },
+    };
+  }
+}
+
+/**
+  * Key Value pair json reference for json Format
+  */
+class JsonFormatRef {
```
Comment (otaviomacedo) [inline packages/aws-cdk-lib/aws-appmesh/lib/shared-interfaces.ts:268]: Why do you need to create this new type?

File: packages/aws-cdk-lib/aws-appmesh/lib/shared-interfaces.ts [hunk 2]
```diff
@@ -143,28 +143,138 @@ class FileAccessLog extends AccessLog {
...
+  }
+}
+
+class TextLoggingFormat extends LoggingFormat {
+  /**
+  * Json pattern for the output logs
+  *
+  * @default - no text format specified
+  */
+  private readonly text?: string;
```
Comment (otaviomacedo) [inline packages/aws-cdk-lib/aws-appmesh/lib/shared-interfaces.ts:250]: Why is this optional?

**Review by otaviomacedo — DISMISSED** [review]

File: packages/aws-cdk-lib/aws-appmesh/lib/shared-interfaces.ts (hunk 2 above)
Comment (EdwardXF) [reply packages/aws-cdk-lib/aws-appmesh/lib/shared-interfaces.ts:250]: The whole field is optional. And you can have either text or json format

File: packages/aws-cdk-lib/aws-appmesh/lib/shared-interfaces.ts (hunk 1 above)
Comment (EdwardXF) [reply packages/aws-cdk-lib/aws-appmesh/lib/shared-interfaces.ts]: Deleted that, that's part of our api design for other parts

File: packages/aws-cdk-lib/aws-appmesh/lib/shared-interfaces.ts (hunk 2 above)
Comment (otaviomacedo) [reply packages/aws-cdk-lib/aws-appmesh/lib/shared-interfaces.ts:245]: Sure, but if want the `text` field in the L1 to be undefined, why bother creating an instance of this class in the first place? In fact, you are already enforcing the text to be defined in the public interface (`LoggingFormat.fromText()`). Do the same here. 

Same for the JSON case.

Comment (EdwardXF) [reply packages/aws-cdk-lib/aws-appmesh/lib/shared-interfaces.ts:245]: Yeah, you are right, being silly here, updated

**Review by otaviomacedo — APPROVED** [review]

**Review by otaviomacedo — APPROVED** [review]

**Pull request merged**